engine/src/main/scala/cromwell/services/EngineMetadataServiceActor.scala

```diff
@@ -23,6 +45,34 @@ case class EngineMetadataServiceActor(serviceConfig: Config, globalConfig: Confi
     }
   }
 
+  private def queryStatusAndRespond(id: WorkflowId): Unit = {
+    val sndr = sender()
+    dataAccess.getWorkflowStatus(id) onComplete {
```

Comment:
@Horneth AFAICT that's fraught with race conditions

@geoffjentry Personally I don't care as much about the ideological purity of separating this stuff as much as UX / not having a stream of tickets filed for this behavior.  😄